src/views/chats/ClosedChats/index.vue

```diff
@@ -0,0 +1,213 @@
+      <unnnic-chats-header
+        v-if="!isLoadingSelectedRoom && selectedRoom"
+        :title="selectedRoom.contact.name"
+        :avatarName="selectedRoom.contact.name"
+      />
+    </header>
+    <main>
+      <section v-if="roomId" class="closed-chats__selected-chat">
+        <room-loading v-show="isLoadingSelectedRoom" only-messages />
+        <chat-messages
+          v-if="selectedRoom"
+          v-show="!isLoadingSelectedRoom"
+          :room="selectedRoom"
+          @scrollTop="chatScrollTop"
+        />
+        <contact-info v-if="selectedRoom" is-history :closedRoom="selectedRoom" @close="() => {}" />
+      </section>
+
+      <closed-chats-rooms-table v-else :project="project" />
+    </main>
+  </div>
+</template>
+
+<script>
+import ProjectApi from '@/services/api/resources/settings/project';
+import History from '@/services/api/resources/chats/history';
+
+import ChatMessages from '@/components/chats/chat/ChatMessages';
+import ContactInfo from '@/components/chats/ContactInfo';
+import ClosedChatsHeaderLoading from '@/views/loadings/ClosedChats/ClosedChatsHeader.vue';
+import RoomHeaderLoading from '@/views/loadings/RoomHeader.vue';
+import RoomLoading from '@/views/loadings/Room.vue';
+import { mapState } from 'vuex';
+import ClosedChatsRoomsTable from './RoomsTable';
+
+export default {
+  name: 'ClosedChats',
+
+  components: {
+    ClosedChatsHeaderLoading,
+    RoomHeaderLoading,
+    RoomLoading,
+    ContactInfo,
+    ClosedChatsRoomsTable,
+    ChatMessages,
+  },
+
+  props: {
+    roomId: {
+      type: String,
+      default: '',
+    },
+  },
+
+  data: () => ({
+    isLoadingHeader: true,
+    isLoadingSelectedRoom: false,
+
+    crumbs: [
+      {
+        name: 'Chats',
+        path: 'home',
+      },
+    ],
+    project: null,
+
+    selectedRoom: null,
+    selectedRoomsUuids: null,
+  }),
+
+  async created() {
+    this.projectInfo();
+    this.crumbs.push({
+      name: this.$t('chats.closed_chats.history'),
+      path: 'closed-rooms',
+    });
+  },
+
+  computed: {
+    ...mapState({
+      roomMessagesNext: (state) => state.roomMessages.roomMessagesNext,
+    }),
+  },
+
+  methods: {
+    backToHome() {
+      this.$router.push({ name: 'home' });
+    },
+
+    handlerCrumbClick(crumb) {
+      if (crumb.name === this.selectedRoom.contact.name) return;
+
+      this.selectedRoom = null;
+
+      const index = this.crumbs.findIndex((item) => item.path === crumb.path);
+      this.crumbs = this.crumbs.slice(0, index + 1);
+
+      this.$router.push({ name: crumb.path });
+    },
+
+    async projectInfo() {
+      const project = await ProjectApi.getInfo();
+      this.project = project.data;
+      this.isLoadingHeader = false;
+    },
+
+    async chatScrollTop() {
+      if (this.roomMessagesNext) {
+        this.getHistoryContactRoomMessages();
+      } else {
+        // const roomUuidIndex = this.selectedRoomsUuids?.findIndex(
```

Comment:
dead code?

Reply:
No, this code is commented to handle future updates with the backend